File: missing_number/perl5/missing_number.pl
#!/usr/bin/perl
use strict;

if (2 != @ARGV) {
	die "usage: $0 <file1> <file2>\n";
}

my ($file1, $file2) = @ARGV;

open(FILE1, "< $file1")
	or die "open '$file1' failed: $!";

open(FILE2, "< $file2")
	or die "open '$file2' failed: $!";

my $sum1 = 0;
my $sum2 = 0;

while (my $line = <FILE1>) {
	chomp($line);
	$sum1 += $line;
}

while (my $line = <FILE2>) {
	chomp($line);
	$sum2 += $line;
}

my $n = ($sum2 > $sum1) ? ($sum2 - $sum1) : ($sum1 - $sum2);

print "$n\n";

close(FILE1);
close(FILE2);

File: missing_number/python3/missing_number.py
#!/usr/bin/python3

import sys

if (len(sys.argv) != 3):
    sys.stderr.write('usage: ' + sys.argv[0] + ' <file1> <file2>\n')
    sys.exit(1)

file1 = open(sys.argv[1], 'r')
file2 = open(sys.argv[2], 'r')

sum1 = 0;
for line in file1:
    sum1 += int(line)

sum2 = 0;
for line in file2:
    sum2 += int(line)

if (sum2 > sum1):
    n = sum2 - sum1
else:
    n = sum1 - sum2

print(n)

file1.close()
file2.close()
